### ISSUE TITLE
Admin account tags leak to non admin accounts

### ISSUE THREAD

**Issue by erdii** [nextcloud/bookmarks#451]
### Steps to reproduce
1. create some tagged bookmarks from account A (admin)
2. log in with account B
3. search for tags on account B

### Expected behaviour
Tell us what should happen

only tags from account B should show up

### Actual behaviour
Tell us what happens instead

tags are mixed with tags from account A (and possibly other accounts, too)

### Server configuration

**Operating system**: docker - nextcloud:13-apache base image

**Web server:** apache

**Database:** docker image: mariadb:10

```
Art: mysql
Version: 10.2.13
```

**PHP version:** PHP 7.1.14

**Nextcloud version:** (see Nextcloud admin page) 13.0.0

**Bookmarks version:** (see Nextcloud apps page) 0.10.1

**Updated from an older Nextcloud or fresh install:** fresh install

**Signing status:**

```
Login as admin user into your Nextcloud and access  
http://example.com/index.php/settings/integrity/failed 
paste the results here.
```

Result: `No errors have been found.`

**List of activated apps:**

* Activity 2.6.1
* Announcement center 3.2.1
* AppOrder 0.4.1
* Bookmarks 0.10.1
* Brute-force settings 1.0.3
* Calendar 1.6.1
* Circles 0.13.6
* Collaborative tags 1.3.0
* Comments 1.3.0
* Contacts 2.1.2
* Deleted files 1.3.0
* External sites 3.0.2
* External user support 0.4
* Federation 1.3.0
* File sharing 1.5.0
* First run wizard 2.2.1
* Gallery 18.0.0
* Impersonate 1.0.2
* Log Reader 2.0.0
* Metadata 0.6.0
* Monitoring 1.3.0
* mood 0.3.3
* Music 0.5.6
* News 12.0.1
* Nextcloud announcements 1.2.0
* Notes 2.3.2
* Notifications 2.1.2
* ONLYOFFICE 1.2.0
* Password policy 1.3.0
* PDF viewer 1.2.0
* Polls 0.8.1
* Quota warning 1.2.0
* Reader (ebook reader) 1.2.3
* Share by mail 1.3.0
* Talk 3.1.0
* Tasks 0.9.6
* Text editor 2.5.1
* Theming 1.4.1
* Two Factor TOTP Provider 1.4.1
* Two Factor U2F 1.5.1
* Update notification 1.3.0
* User account deletion 0.0.7
* Versions 1.6.0
* Video player 1.2.0

```
If you have access to your command line run e.g.:
sudo -u www-data php occ app:list
from within your Nextcloud installation folder
```

**Nextcloud configuration:**

```
If you have access to your command line run e.g.:
sudo -u www-data php occ config:list system
from within your instance's installation folder
```

```JSON
{
    "system": {
        "trusted_proxies": "***REMOVED SENSITIVE VALUE***",
        "overwritehost": "***REMOVED SENSITIVE VALUE***",
        "overwriteprotocol": "https",
        "htaccess.RewriteBase": "\/",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "apps_paths": [
            {
                "path": "\/var\/www\/html\/apps",
                "url": "\/apps",
                "writable": false
            },
            {
                "path": "\/var\/www\/html\/custom_apps",
                "url": "\/custom_apps",
                "writable": true
            }
        ],
        "instanceid": "***REMOVED SENSITIVE VALUE***",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "***REMOVED SENSITIVE VALUE***"
        ],
        "datadirectory": "***REMOVED SENSITIVE VALUE***",
        "overwrite.cli.url": "***REMOVED SENSITIVE VALUE***",
        "dbtype": "mysql",
        "version": "13.0.0.14",
        "dbname": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "***REMOVED SENSITIVE VALUE***",
        "dbport": "",
        "dbtableprefix": "oc_",
        "mysql.utf8mb4": true,
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "user_backends": [
            {
                "class": "\\OCA\\User_External\\WebDAVAuth",
                "arguments": [
                    "***REMOVED SENSITIVE VALUE***"
                ]
            }
        ],
        "loglevel": 4,
        "logfile": "\/var\/www\/html\/config\/nextcloud.log",
        "mail_smtpmode": "smtp",
        "mail_smtpauthtype": "LOGIN",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpsecure": "ssl",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "465",
        "mail_smtpauth": 1,
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "memcache.distributed": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "***REMOVED SENSITIVE VALUE***",
            "port": "6379"
        },
        "maintenance": false
    }
}
```

Make sure to remove all sensitive content such as passwords. (e.g. database password, passwordsalt, secret, smtp password, …)

**Are you using external storage, if yes which one:** no

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** Webdav

### Client configuration
**Browser:** All browser, current version

**Operating system:** tested on windows / mac os / linux

#### Web server error log
no relevant errors

#### Nextcloud log (data/nextcloud.log)
no relevant errors

#### Browser log
GET https://REDACTED/apps/bookmarks/bookmark?type=rel_tags&tag=test&page=0&sort=bookmarks_sorting_recent

response:

```JSON
{
  "data": [
    {
      "tag": "coding",
      "nbr": "7"
    },
    {
      "tag": "utils",
      "nbr": "4"
    },
    {
      "tag": "nice",
      "nbr": "3"
    },
    {
      "tag": "css",
      "nbr": "3"
    },
    {
      "tag": "bash",
      "nbr": "3"
    },
    {
      "tag": "reads",
      "nbr": "3"
    },
    {
      "tag": "privacy",
      "nbr": "2"
    },
    {
      "tag": "tweaks",
      "nbr": "2"
    },
    {
      "tag": "design",
      "nbr": "2"
    },
    {
      "tag": "productivity",
      "nbr": "2"
    },
    {
      "tag": "p2p",
      "nbr": "2"
    },
    {
      "tag": "devops",
      "nbr": "2"
    },
    {
      "tag": "office",
      "nbr": "1"
    },
    {
      "tag": "free",
      "nbr": "1"
    },
    {
      "tag": "tricks",
      "nbr": "1"
    },
    {
      "tag": "management",
      "nbr": "1"
    },
    {
      "tag": "dns",
      "nbr": "1"
    },
    {
      "tag": "webdav",
      "nbr": "1"
    },
    {
      "tag": "bug",
      "nbr": "1"
    },
    {
      "tag": "dos",
      "nbr": "1"
    },
    {
      "tag": "geeky",
      "nbr": "1"
    },
    {
      "tag": "debugging",
      "nbr": "1"
    },
    {
      "tag": "shell",
      "nbr": "1"
    },
    {
      "tag": "crypto",
      "nbr": "1"
    },
    {
      "tag": "kochen",
      "nbr": "1"
    },
    {
      "tag": "encryption",
      "nbr": "1"
    },
    {
      "tag": "gpg",
      "nbr": "1"
    },
    {
      "tag": "linux",
      "nbr": "1"
    },
    {
      "tag": "engineering",
      "nbr": "1"
    },
    {

```

**Comment by marcelklehr** [comment]
Hey there,

thanks for reporting this! :)

It's now fixed in [v0.11.0](https://github.com/nextcloud/bookmarks/releases/tag/v0.11.0) ;)

(Also: duplicate of #448, in case you're wondering how we pulled this off so fast :D)

Shout, if it's not fixed for you, though.

Cheers!

**Comment by erdii** [comment]
Ah! Cool, thank you for your fast reply!
Thats' why the query to fetch the tags didn't look suspicious at all. 😅

Thanks for making this app happen! 🙌
